### COVID-19 Analysis Project

In [7]:
# !pip install plotly pandas matplotlib numpy

In [9]:
# Imoporting Required Libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import plotly.express as px

In [10]:
# correct the renderer for plotly graphs
import plotly.io as pio
pio.renderers.default = "iframe"

In [12]:
df = pd.read_csv('country_wise_latest.csv')
df.head()

Country/Region  Confirmed  Deaths  Recovered  Active  New cases  New deaths  \
0    Afghanistan      36263    1269      25198    9796        106          10   
1        Albania       4880     144       2745    1991        117           6   
2        Algeria      27973    1163      18837    7973        616           8   
3        Andorra        907      52        803      52         10           0   
4         Angola        950      41        242     667         18           1   

   New recovered  Deaths / 100 Cases  Recovered / 100 Cases  \
0             18                3.50                  69.49   
1             63                2.95                  56.25   
2            749                4.16                  67.34   
3              0                5.73                  88.53   
4              0                4.32                  25.47   

   Deaths / 100 Recovered  Confirmed last week  1 week change  \
0                    5.04                35526            737   
1                    5.25                 4171            709   
2                    6.17                23691           4282   
3                    6.48                  884             23   
4                   16.94                  749            201   

   1 week % increase             WHO Region  
0               2.07  Eastern Mediterranean  
1              17.00                 Europe  
2              18.07                 Africa  
3               2.60                 Europe  
4              26.84                 Africa

what I have undersatnd from this data:

confirmed = death + recovered + active
Active --> the number of patient currenly in each country

WHO Region --> refers to the World Health Organization regional classification

Deaths / 100 cases--> the case fatality rate (CFR) (how deadly the virus is in each country) expressed per 100 cases.

Recovered / 100 cases --> Shows how many people recovered out of every 100 who were infected. (how well each country recovers infected patients)

Deaths / 100 recovered --> Shows how deaths relate to recoveries it's a measure of outcome severity. (high Deaths / 100 Recovered = more deaths per recovery, showing poor medical outcome)

In [13]:
# check the dataframe information and data types
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 187 entries, 0 to 186
Data columns (total 15 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Country/Region          187 non-null    object 
 1   Confirmed               187 non-null    int64  
 2   Deaths                  187 non-null    int64  
 3   Recovered               187 non-null    int64  
 4   Active                  187 non-null    int64  
 5   New cases               187 non-null    int64  
 6   New deaths              187 non-null    int64  
 7   New recovered           187 non-null    int64  
 8   Deaths / 100 Cases      187 non-null    float64
 9   Recovered / 100 Cases   187 non-null    float64
 10  Deaths / 100 Recovered  187 non-null    float64
 11  Confirmed last week     187 non-null    int64  
 12  1 week change           187 non-null    int64  
 13  1 week % increase       187 non-null    float64
 14  WHO Region              187 non-null    ob

In [14]:
# check if there is a duplicated rows
df.duplicated().sum()

np.int64(0)

There are no null values or duplicated rows. Let's see the numeric values description

In [15]:
df.describe()

C:\Users\THIS\AppData\Local\Programs\Python\Python312\Lib\site-packages\pandas\core\nanops.py:1016: RuntimeWarning:

invalid value encountered in subtract



Confirmed         Deaths     Recovered        Active     New cases  \
count  1.870000e+02     187.000000  1.870000e+02  1.870000e+02    187.000000   
mean   8.813094e+04    3497.518717  5.063148e+04  3.400194e+04   1222.957219   
std    3.833187e+05   14100.002482  1.901882e+05  2.133262e+05   5710.374790   
min    1.000000e+01       0.000000  0.000000e+00  0.000000e+00      0.000000   
25%    1.114000e+03      18.500000  6.265000e+02  1.415000e+02      4.000000   
50%    5.059000e+03     108.000000  2.815000e+03  1.600000e+03     49.000000   
75%    4.046050e+04     734.000000  2.260600e+04  9.149000e+03    419.500000   
max    4.290259e+06  148011.000000  1.846641e+06  2.816444e+06  56336.000000   

        New deaths  New recovered  Deaths / 100 Cases  Recovered / 100 Cases  \
count   187.000000     187.000000          187.000000             187.000000   
mean     28.957219     933.812834            3.019519              64.820535   
std     120.037173    4197.719635            3.454302              26.287694   
min       0.000000       0.000000            0.000000               0.000000   
25%       0.000000       0.000000            0.945000              48.770000   
50%       1.000000      22.000000            2.150000              71.320000   
75%       6.000000     221.000000            3.875000              86.885000   
max    1076.000000   33728.000000           28.560000             100.000000   

       Deaths / 100 Recovered  Confirmed last week  1 week change  \
count                  187.00         1.870000e+02     187.000000   
mean                      inf         7.868248e+04    9448.459893   
std                       NaN         3.382737e+05   47491.127684   
min                      0.00         1.000000e+01     -47.000000   
25%                      1.45         1.051500e+03      49.000000   
50%                      3.62         5.020000e+03     432.000000   
75%                      6.44         3.708050e+04    3172.000000   
max                       inf         3.834677e+06  455582.000000   

       1 week % increase  
count         187.000000  
mean           13.606203  
std            24.509838  
min            -3.840000  
25%             2.775000  
50%             6.890000  
75%            16.855000  
max           226.320000

The warning appeared because there are some inf numbers in the (Deaths / 100 Recovered) column. So let's print them and see why there is such an answer

In [16]:
df[df['Deaths / 100 Recovered']== np.inf]

Country/Region  Confirmed  Deaths  Recovered  Active  New cases  \
32          Canada     116458    8944          0  107514        682   
117     Mozambique       1701      11          0    1690         32   
147         Serbia      24141     543          0   23598        411   
161         Sweden      79395    5700          0   73695        398   
163          Syria        674      40          0     634         24   

     New deaths  New recovered  Deaths / 100 Cases  Recovered / 100 Cases  \
32           11              0                7.68                    0.0   
117           0              0                0.65                    0.0   
147           9              0                2.25                    0.0   
161           3              0                7.18                    0.0   
163           2              0                5.93                    0.0   

     Deaths / 100 Recovered  Confirmed last week  1 week change  \
32                      inf               112925           3533   
117                     inf                 1507            194   
147                     inf                21253           2888   
161                     inf                78048           1347   
163                     inf                  522            152   

     1 week % increase             WHO Region  
32                3.13               Americas  
117              12.87                 Africa  
147              13.59                 Europe  
161               1.73                 Europe  
163              29.12  Eastern Mediterranean

So for those countries, there are no recovered cases yet, which is why the Deaths / 100 Recovered(outcome severity) rate is extremely high(infinite number)

In [17]:
# object description
df.describe(include='O')

Country/Region WHO Region
count             187        187
unique            187          6
top       Afghanistan     Europe
freq                1         56

So the number of unique countries is 187, and there are 6 different regions of the World Health Organization Regions (continent) column

In [18]:
#the number of countries in each region
regions = df['WHO Region'].value_counts().reset_index()
fig = px.bar(regions, x='WHO Region' , y='count', labels={'WHO Region':'World Health Organization Regions'},title='Number of Countries in Each Region')
fig.show()

In [19]:
# country that has the largest death number
df[['Country/Region','Deaths']].groupby(['Country/Region']).max().sort_values(by='Deaths',ascending=False).head(1)

Deaths
Country/Region        
US              148011

In [20]:
# check the country which has the highest recovered people number
df.loc[df['Recovered'] == df['Recovered'].max(), 'Country/Region']

23    Brazil
Name: Country/Region, dtype: object

In [21]:
# the countries that has the lowest death number wich is 0
# we have seen(in the describe function) the minimum values of column death which is zero
df.loc[df['Deaths'] == 0, 'Country/Region']

19                               Bhutan
30                             Cambodia
49                             Dominica
55                              Eritrea
59                                 Fiji
68                            Greenland
69                              Grenada
75                             Holy See
94                                 Laos
114                            Mongolia
130                    Papua New Guinea
140               Saint Kitts and Nevis
141                         Saint Lucia
142    Saint Vincent and the Grenadines
148                          Seychelles
168                         Timor-Leste
181                             Vietnam
Name: Country/Region, dtype: object

In [22]:
# the country which has the lowest recovered people number
# we have seen(in the describe function) the minimum values of column recovered which is zero
df.loc[df['Recovered'] == df['Recovered'].min(), ['Country/Region','Recovered']]

Country/Region  Recovered
32          Canada          0
117     Mozambique          0
147         Serbia          0
161         Sweden          0
163          Syria          0
168    Timor-Leste          0

In [24]:
# Top 5 countries that have the highest active cases now
contries = df.groupby('Country/Region')['Active'].sum().sort_values( ascending = False).head().reset_index()

fig = px.bar(contries, x = 'Country/Region',y='Active', title='Top 5 Countries that Have the Highest Active Cases', labels={'Country/Region':'Countries'})
fig.show()

In [25]:
# the world map to represent the intensity of Confirmed cases in each country
fig = px.choropleth(df, locations='Country/Region',color='Confirmed',hover_name='Country/Region',locationmode="country names", color_continuous_scale=px.colors.sequential.Sunsetdark, title='Confirmed Cases In Each Countries')
fig.show()

C:\Users\THIS\AppData\Local\Temp\ipykernel_9720\2379303275.py:2: DeprecationWarning:

The library used by the *country names* `locationmode` option is changing in an upcoming version. Country names in existing plots may not work in the new version. To ensure consistent behavior, consider setting `locationmode` to *ISO-3*.



We can see the distribution of the confirmed cases in each country and how America, Brazil, India, and Russia have the most cases between active, recovered, and deaths

In [26]:
# the death rate in precentage in each region
continent_summary = df.groupby('WHO Region')[['Deaths', 'Confirmed']].sum()
continent_summary['Death Rate %'] = (continent_summary['Deaths'] / continent_summary['Confirmed']) * 100

fig = px.bar(continent_summary ,y='Deaths',labels={'WHO Region':'World Health Organization Regions'} ,title='World Health Organization Regions that Have the Higher Death Rate')
fig.show()

We can see that America and Europe have the highest death rates

In [27]:
# the active rate in precentage in each region
continent_summary = df.groupby('WHO Region')[['Active', 'Confirmed']].sum()
continent_summary['Death Rate %'] = (continent_summary['Active'] / continent_summary['Confirmed']) * 100

fig = px.bar(continent_summary ,y='Active', labels={'WHO Region':'World Health Organization Regions'} ,title='World Health Organization Regions That Have The Higher Active Rate')
fig.show()

We can see that America and Europe have the highest active cases, and Africa has more active cases than the Eastern Mediterranean, whereas in the death rate, the Eastern Mediterranean has the highest death rate

In [28]:
# top 10 countries that have the higher case fatality rate
higher_case_fatality = df.sort_values(by=['Deaths / 100 Cases'],ascending=False).head(10)
fig = px.bar(higher_case_fatality, x='Country/Region',y='Deaths / 100 Cases',color="WHO Region",title="Deaths per 100 COVID-19 Cases by Country",labels={"Deaths / 100 Cases": "Deaths per 100 Cases", 'Country/Region':'Countries'} ,hover_data=['Deaths','Active'])
fig.show()

The top 10 countries in the case fatality rate (CFR) per 100 cases across regions indicates how deadly the virus is in each country. We can see that Europe has 7 countries in this list, which can reflect a healthcare system stress in these countries, thus the high CFR rate.

In [29]:
# top 10 countries that have the higher recovery efficiency
recovery_efficiency = df.sort_values(by=['Recovered / 100 Cases'],ascending=False).head(10)
fig = px.bar(recovery_efficiency, x='Country/Region',y='Recovered / 100 Cases',color="WHO Region" ,hover_data=['Recovered','Active'],title="Recovered per 100 COVID-19 Cases by Country",
    labels={"Recovered / 100 Cases": "Recovered per 100 Cases",'Country/Region':'Countries'})
fig.show()

The top 10 countries in recovery efficiency, which indicates how well each country recovers infected patients across regions. We can see that there is no specific region that has a significantly higher recovery rate by country, but the Western Pacific region has 3 countries.

In [30]:
continent_summary = df.groupby('WHO Region')[['Deaths', 'Recovered','Active','Confirmed']].sum().reset_index()

# Melting to long format
df_long = continent_summary.melt(id_vars='WHO Region', value_vars=['Recovered','Deaths','Active','Confirmed'],var_name='Status', value_name='Count')

fig = px.bar(df_long,
             x='WHO Region', y='Count', color='Status',
             barmode='group',  # side by side bars
             title='COVID-19 Cases by WHO Region and Status',
             labels={'Count': 'Number of Cases', 'WHO Region': 'WHO Region'})
fig.update_yaxes(type="log")
fig.show()

comparing the different regions according to the active, recovered, death, and confirmed cases. We can tell that the virus is widely spread in the Americas region, hence the high numbers in all the status, unlike the Western Pacific region

In [31]:
# countries with high new Deaths and low new recoveries
fig = px.scatter(df, x="New recovered", y="New deaths", color="WHO Region", hover_name="Country/Region", size="Confirmed", title="Countries with High New Deaths and Low New Recoveries")
fig.show()

The countries (like Peru, Mexico) in the upper left position have high new deaths and low new recoveries, which is a warning sign that the situation in such countries is worsening, the upper left position have high new deaths and low new recoveries, which is a warning sign that the situation in such countries is worsening and may be the healthcare system is falling.

In [32]:
# focusing on the coutries with critical conditions

avg_new_death = df['New deaths'].mean()
avg_new_recovered = df['New recovered'].mean()

critical = df[(df['New deaths'] > avg_new_death) & (df['New recovered'] < avg_new_recovered)]
px.bar(critical, x='Country/Region', y='New deaths', color='WHO Region', labels={'Country/Region':'Countries'})

Those two countries are considered critical conditions because they have above-average new deaths, and the new recoveries are below average in each of them are in the Americas region.

In [33]:
# High 1 week % increase (speed of the outbreak) with high Active (size of the outbreak)→ outbreak growing
# Finds fast-growing and heavily affected countries
avg_increase = df['1 week % increase'].mean()
avg_active = df['Active'].mean()
heavily_affected_countries = df[(df['1 week % increase'] > avg_increase) & (df['Active'] > avg_active)]

fig = px.scatter(heavily_affected_countries, x='1 week % increase',
y='Active', size='Confirmed', color='WHO Region', hover_name='Country/Region', title='COVID-19 Hotspots: High Growth and High Active Cases')
fig.show()

Identifying heavily affected countries where both 1-week percentage increases and the active cases are above average, those countries are identified as hotspots with fast-growing outbreaks, and many people are currently infected across the region. We can see that India has the highest increase in one week, with the highest active cases.

#### Project End!